### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by SalahAdDin**

### CODE REVIEW

File: app/templates/app/category_detail.html [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{% extends 'layout/base.html' %}
+
+{% block content %}
+    <h1>
+        {{ category.name }}
+    </h1>
+    <ul>
+        {% for article in category.article_set.all %}
```
Comment (camilonova) [inline app/templates/app/category_detail.html]: Ident to the left

File: app/templates/app/category_detail.html [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{% extends 'layout/base.html' %}
+
+{% block content %}
+    <h1>
```
Comment (camilonova) [inline app/templates/app/category_detail.html]: Leave it all in one line

File: app/templates/app/article_detail.html [hunk 3]
```diff
@@ -0,0 +1,17 @@
+{% extends 'layout/base.html' %}
+
+{% block content %}
+    <h1>
+        {{ article.name }}
+        <small>{{ article.updated_at }}</small>
+    </h1>
+    <p>
```
Comment (camilonova) [inline app/templates/app/article_detail.html]: Use an article instead of a p

File: app/templates/home.html [hunk 4]
```diff
@@ -1,5 +1,20 @@
...
 
 {% block content %}
-<h1>Hola</h1>
+    {% include '_search_box.html' %}
+    {% for category in category_list %}
+        <h1>
+            <a href="{% url 'category_detail' category.slug %}">{{ category.name }}</a>
+        </h1>
+        <ul>
+            {% for article in category.active_articles_list|slice:":5" %}
```
Comment (camilonova) [inline app/templates/home.html]: Ident whole block to the left

File: app/templates/home.html [hunk 5]
```diff
@@ -1,5 +1,20 @@
...
+                <li>
+                    <a href="{% url 'article_detail' article.slug %}">{{ article.name }}</a>
+                </li>
+            {% endfor %}
+        </ul>
+        {% if category.active_articles_list.count > 5 %}
+            <a href="{% url 'category_detail' category.slug %}">Ver todos</a>
```
Comment (camilonova) [inline app/templates/home.html:17]: wrap this i a <p>

File: app/views.py [hunk 6]
```diff
@@ -1,5 +1,36 @@
...
+from .models import Category
+from .models import Article
+from .forms import SearchForm
 
-class HomeView(TemplateView):
+
+class HomeView(ListView):
     template_name = 'home.html'
+    model = Category
+    context_object_name = 'category_list'
```
Comment (camilonova) [inline app/views.py]: Don't set this. Use `object_list` in the template

File: app/views.py [hunk 7]
```diff
@@ -1,5 +1,36 @@
...
+
+    def get_context_data(self, **kwargs):
+        context = super(HomeView, self).get_context_data(**kwargs)
+        context['form'] = SearchForm()
+        return context
+
+
+class CategoryDetailView(DetailView):
+    model = Category
+    context_object_name = 'category'
```
Comment (camilonova) [inline app/views.py]: Don't set this. Use `object` in the template

File: app/views.py [hunk 8]
```diff
@@ -1,5 +1,36 @@
...
+
+
+class ArticleDetailView(DetailView):
+    model = Article
+    context_object_name = 'article'
```
Comment (camilonova) [inline app/views.py]: Same here

File: app/templates/app/search_results_list.html [hunk 9]
```diff
@@ -0,0 +1,18 @@
+{% extends 'layout/base.html' %}
+
+{% block content %}
+    {% include '_search_box.html' %}
+
+    {% for article in object_list %}
+        <h1>
```
Comment (PabloVallejo) [inline app/templates/app/search_results_list.html]: Leave it in one line `<h1>{{ article.name }}</h1>`

**Pull request merged**